hudi-client/hudi-flink-client/src/main/java/org/apache/hudi/table/action/compact/HoodieFlinkMergeOnReadTableCompactor.java

```diff
@@ -60,4 +79,19 @@ public void preCompact(
   public void maybePersist(HoodieData<WriteStatus> writeStatus, HoodieEngineContext context, HoodieWriteConfig config, String instantTime) {
     // No OP
   }
+
+  @Override
+  public List<WriteStatus> compact(HoodieCompactionHandler compactionHandler,
+                                   HoodieTableMetaClient metaClient,
+                                   HoodieWriteConfig writeConfig,
+                                   CompactionOperation operation,
+                                   String instantTime,
+                                   Option<EngineBroadcastManager> broadcastManagerOpt) throws IOException {
+    Configuration conf = metaClient.getStorage().getConf().unwrapAs(Configuration.class);
+    FlinkRowDataReaderContext readerContext = new FlinkRowDataReaderContext(
```

Comment:
Looks like there is no need to store Flink conf in `FlinkRowDataReaderContext`, the `InternalSchemaManager` needs the Flink conf to do 2 things:

1. decide if the schema evolution is enabled: we can move this check into `HoodieWriteConfig.isSchemaEvolutionEnabled`;
2. generates Hadoop conf from it, but the hadoop conf is already in `FlinkRowDataReaderContext`.

Reply:
Besides the usage you mentioned, flink conf stored in `FlinkRowDataReaderContext` is also used to:
* generate partition specs for `FlinkParquetReader`;
* get config `read.utc-timezone` to create field converter (Flink value -> Avro value) in `getOrderingValue`;

Any recommended cleaner way to achieve these?